guides/06_client-libraries/02_getting-started-with-the-js-client.md

```diff
@@ -0,0 +1,251 @@
+const transcription = app.predict("/predict", [audio_file]);
+```
+
+If you have previously duplicated a Space, re-running `duplicate` will *not* create a new Space. Instead, the client will attach to the previously-created Space. So it is safe to re-run the `duplicate` method multiple times with the same space. 
+
+**Note:** if the original Space uses GPUs, your private Space will as well, and your Hugging Face account will get billed based on the price of the GPU. To minimize charges, your Space will automatically go to sleep after 1 hour of inactivity. You can also set the hardware using the `hardware` parameter of `duplicate()`.
+
+
+## Connecting a general Gradio app
+
+If your app is running somewhere else, just provide the full URL instead, including the "http://" or "https://". Here's an example of making predictions to a Gradio app that is running on a share URL:
+
+```js
+import { client } from "@gradio/client";
+
+const app = client("https://bec81a83-5b5c-471e.gradio.live");
+```
+
+## Inspecting the API endpoints
+
+Once you have connected to a Gradio app, you can view the APIs that are available to you by calling the `client`'s `view_api` method. For the Whisper Space, we see the following:
+
+```json
+{
+  "named_endpoints": {
+    "/predict": {
+      "parameters": [
+        {
+          "label": "text",
+          "component": "Textbox",
+          "type": "string"
+        }
+      ],
+      "returns": [
+        {
+          "label": "output",
+          "component": "Textbox",
+          "type": "string"
+        }
+      ]
+    }
+  },
+  "unnamed_endpoints": {}
+}
+```
+
+This shows us that we have 1 API endpoint in this space, and shows us how to use the API endpoint to make a prediction: we should call the `.predict()` method (which we will explore below), providing a parameter `input_audio` of type `string`, which is a url to a file. 
+
+We should also provide the `api_name='/predict'` argument to the `predict()` method. Although this isn't necessary if a Gradio app has only 1 named endpoint, it does allow us to call different endpoints in a single app if they are available. If an app has unnamed API endpoints, these can also be displayed by running `.view_api(all_endpoints=True)`.
+
+
+## Making a prediction
+
+The simplest way to make a prediction is simply to call the `.predict()` method with the appropriate arguments:
+
+```js
+import { client } from "@gradio/client";
+
+const app = await client("abidlabs/en2fr");
+const result = await client.predict("/predict", ["Hello"]);
+```
+
+If there are multiple parameters, then you should pass them as an array to `.predict()`, like this:
+
+```js
+import { client } from "@gradio/client";
+
+const app = await client("gradio/calculator");
+const result = await client.predict("/predict", [4, "add", 5]);
+```
+
+For certain inputs, such as images, you should pass in a `Buffer`, `Blob` or `File` depending on what is most convenient and what environment the client is running in. 
```

Comment:
What do you mean by this? How does it depend on the environment? Will any of these 3 choices work?

Reply:
Node vs Browser basically. All 3 work but which one you use would be dictated by the environment.

I'll clarify this to:
```suggestion
For certain inputs, such as images, you should pass in a `Buffer`, `Blob` or `File` depending on what is most convenient. In node, this would be a `Buffer` or `Blob`; in a browser environment, this would be a `Blob` or `File`.
```